File: atcoder.jp/contests/abc287/tasks/abc287_b/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
  int N, M;
  cin >> N >> M;
  vector<string> S(N), T(M);
  for (auto &&v : S) cin >> v;
  for (auto &&v : T) cin >> v;

  set<string> st = {T.begin(), T.end()};

  int ans = 0;
  for (auto &&s : S)
    if (st.count(s.substr(3))) ans++;

  cout << ans << endl;
}

File: atcoder.jp/contests/abc287/tasks/abc287_b/main.rs
use std::collections::HashSet;

use proconio::input;

fn main() {
    input! {
        n: usize,
        m: usize,
        s: [String; n],
        t: [String; m],
    }

    let hs: HashSet<_> = t.into_iter().collect();
    let ans = s.into_iter().filter(|s| hs.contains(&s[3..])).count();

    println!("{}", ans);
}
